runtime/src/main/cpp/TypeInfo.h

```diff
@@ -114,6 +114,9 @@ struct TypeInfo {
     WritableTypeInfo* writableInfo_;
 #endif
 
+    // Null-terminated array of pairs (TypeInfo*, OBJ_GETTER0(getAssociatedObjectInstance)).
+    void** associatedObjects;
```

Comment:
Why not type properly?

Reply:
Because checking for null terminator becomes somewhat dirty in this case.
We get forced either to including extra pointer or making last element in this array half-defined. Do you find this ok?

Reply:
Other similar structures in this type have explicit count next to it. We do it this way, or just zero TypeInfo in the terminator. I probably like the first one more.

Reply:
The first approach applied the same way as for other data would make every TypeInfo larger. We can use additional indirection here, but using zero terminator looks much simpler to me.